fpm/app/main.f90

```diff
@@ -1,27 +1,31 @@
 program main
-use fpm, only: print_help, cmd_build, cmd_install, cmd_new, cmd_run, cmd_test
+use command_line, only: &
+        fpm_cmd_settings, &
+        fpm_new_settings, &
+        fpm_build_settings, &
+        fpm_run_settings, &
+        fpm_test_settings, &
+        fpm_install_settings, &
+        get_command_line_settings
+use fpm, only: cmd_build, cmd_install, cmd_new, cmd_run, cmd_test
+
 implicit none
-character(100) :: cmdarg
 
-if (command_argument_count() == 0) then
-    call print_help()
-else if (command_argument_count() == 1) then
-    call get_command_argument(1, cmdarg)
-    select case(trim(cmdarg))
-        case("build")
-            call cmd_build()
-        case("install")
-            call cmd_install()
-        case("new")
-            call cmd_new()
-        case("run")
-            call cmd_run()
-        case default
-            print *, "fpm error: No such command " // trim(cmdarg)
-            error stop 1
-    end select
-else
-    print *, "Too many arguments"
-    error stop 1
-end if
+class(fpm_cmd_settings), allocatable :: cmd_settings
+
+call get_command_line_settings(cmd_settings)
+
+select type(cmd_settings)
```

Comment:
Why cannot this just be an integer? It seems quite complicated this way with the class.

Reply:
The different commands will have different options and arguments associated with them. This way the various operations are essentially completely independent.

Reply:
I'm not a fan of this style either, IMO it unnecessarily complicates the logic. You'll have as much boilerplate as with a procedural style. But it works so I'm not opposed to it.